openedx_learning/apps/core/itemstore/models.py

```diff
@@ -0,0 +1,133 @@
+"""
+The item model hiearachy is: Item -> ItemVersion -> ItemInfo -> ItemRaw
+
+Item is the versionless thing that is guaranteed to exist for the lifetime of
+the LearningContext. An ItemVersion is a different version of that item for a
+given LearningContext, and may include policy changes (like grading). ItemInfo
+covers basic metadata that is intrinsic to the item itself, and now how it's
+used in a LearningContext. ItemRaw represents the raw byte data.
+
+TODO: Add link to ADR after it merges.
+"""
+from django.db import models
+
+from openedx_learning.lib.fields import (
+    hash_field,
+    identifier_field,
+    immutable_uuid_field,
+    manual_date_time_field,
+)
+from ..publish.models import LearningContext, LearningContextVersion
+
+
+class Item(models.Model):
+    """
+    This represents any content that has ever existed in a LearningContext.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
+    identifier = identifier_field()
+
+    # TODO: These pointers to the latest published version are convenient, but
+    # we don't currently have the data integrity guarantees to make sure that 
+    # multiple versions aren't active at the same time. Maybe add a ref to the
+    # Item model in LearningContextVersionItemVersion, so we can force that
+    # constraint?
+    published_version = models.ForeignKey('ItemVersion', on_delete=models.RESTRICT, null=True, related_name="+")
```

Comment:
knowledge check: this is secondary source of information, since one could authoritatively determine this by asking the `publish` app what the published LearningContextVersion is, and then looking at the ItemVersions within that LCV, right?

Reply:
Yes. And maybe it's not a great idea to have redundancy like this. The main motivation for this was that a lot of the time, you want to bind data against the abstract notion of a piece of content across all its versions (e.g. most student state), and I wanted a really convenient way to get to the current "live" version from there.

Reply:
That does sound convenient. I guess, you're either going to end up needing to keep this field up-to-date, or you're going to end up with a lot of requests making the same query ("what's the published version?"). I don't think I have the experience to know which of these is an easier challenge to manage.

Reply:
What about making `published_version` a [memoized] python getter that uses the publish app to return the published version? Or do you mean specifically you want to be able to use the django ORM methods on this `published_version` field so you can `select_related` and do things like filtering queries?

Reply:
> Or do you mean specifically you want to be able to use the django ORM methods on this published_version field so you can select_related and do things like filtering queries?

Yeah, exactly this. I want to make sure we can still do efficient filtering queries. But I think I'll actually leave this out for now and add it in later if it becomes necessary. Even if we do keep this kind of a key, it might make more sense in a separate model that specifically separates the publishing details (`published_version`, `first_published_at`, `last_published_at`, etc.).